midealocal/devices/e2/__init__.py

```diff
@@ -113,12 +113,10 @@ def set_attribute(self, attr: str, value: Any) -> None:
             DeviceAttributes.keep_warm,
             DeviceAttributes.current_temperature,
         ]:
-            old_protocol: str | bool | None = None
             if self._old_protocol is not None and self._old_protocol != "auto":
                 old_protocol = self._old_protocol
             else:
-                old_protocol = self.old_protocol()
-                raise TypeError("old_protocol is <bool> instead of <str>")
+                old_protocol = "true" if self.old_protocol() else "false"

```

Comment:
"false" will pass the following `if` ...

Reply:
True, reworked the entire logic